server/spec/activation_key_spec.rb

```diff
@@ -15,8 +15,8 @@
     mallory = create_owner random_string('test_owner')
     @mallory_client = user_client(mallory, random_string('testuser'))
 
-    create_pool_and_subscription(@owner['key'], @product.id, 37, [], '', '', '', nil, nil, true)
-    create_pool_and_subscription(@owner['key'], @product2.id, 37)
+    @cp.create_pool(@owner['key'], @product.id, {:quantity => 37})
```

Comment:
@mstead thoughts?

Reply:
I'll try to address each one in order:

- True "mode agnostic" is kind of a pipe dream. In situations where it matters (refresh and manifest import), we need to be explicit on where the data is created and how it's handled, otherwise the tests aren't clear about what is being tested, where the data is coming from, etc. This was the main problem I sought to correct after running into several issues with refresh operations in the past.
For the other tests, they are already implicitly mode agnostic. As it is in this PR, they create a pool "locally," without the need for a refresh because the refresh operation isn't significant to the test, as that's not part of what's being tested. These tests work fine in both modes without the overhead of doing a refresh every time.

- create_pool_and_subscription was hiding a lot of issues. I feel tests need to be explicit about what data they're creating, where they're creating it and how it's being created.
  - When you discuss "hosted mode," what exactly are you referencing? CP doesn't have much difference between the modes and regardless of how the pools are being created, "hosted mode" is still being tested. Literally the only difference here is the how data is being created. With the create_pool_and_subscription, every pool was created by creating a subscription and then refreshing. This in itself doesn't add value to the test (except in the refresh tests), so we're not gaining anything by hiding details behind it.
  - Indeed, test writers need to care about the way data is created for their tests. This is not a bad thing, in my opinion. Someone writing a test should know how their data is created and how it flows through the operations. A general description of a test is: create input data, do operation, verify output data. This does not need to be abstracted.

- In standalone mode we *did* and *do* create pools with the API, even with the old CPaS method. The only other way to create pools in standalone mode is with a manifest import (which is basically just a refresh). In fact, one major issue uncovered by this change was that we weren't testing any of the custom pool paths because all pools were being created as a mock "upstream"-sourced pool.